### PR TITLE
Add precipitation prognostic variables

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR adds precipitation prognostic variables

This is a peel off from #916.

### CODE REVIEW

File: src/InitialConditions/baro_wave.jl [hunk 1]
```diff
@@ -129,10 +129,30 @@ function center_initial_condition_baroclinic_wave(
     end
     # TODO: Include ability to handle nonzero initial cloud condensate
 
+    precip_kwargs =
+        if precip_model isa NoPrecipitation &&
```
Comment (trontrytel) [inline src/InitialConditions/baro_wave.jl:133]: If we use 0-moment microphysics we also should not add q_rai and q_sno to variable list

File: src/InitialConditions/baro_wave.jl [hunk 2]
```diff
@@ -129,10 +129,30 @@ function center_initial_condition_baroclinic_wave(
...
+           !(turbconv_model isa TC.EDMFModel)
+            NamedTuple()
+        else
+            (; q_rai = FT(0), q_sno = FT(0))
```
Comment (trontrytel) [inline src/InitialConditions/baro_wave.jl:137]: Should our tracers for rain be the same as other tracers in the dycore. I.e. be rho * q_something ?

Comment (charleskawczynski) [reply src/InitialConditions/baro_wave.jl:137]: Probably. This is just a peel off from #916, but it's introducing a pretty major performance penalty due to inference failure

Comment (trontrytel) [reply src/InitialConditions/baro_wave.jl:137]: :/ Does that mean we don't want to merge it? Do you have any ideas how to help the compiler there?

Comment (charleskawczynski) [reply src/InitialConditions/baro_wave.jl:137]: Not as is, it's slowing some jobs down _significantly_. I think the best course of action is to somehow make this optional, and merge it while only adding a single job that exercises it (maybe the perf job?). This way, it doesn't impact everyone, but we can keep an eye on / experiment with it.

**Comment by charleskawczynski** [comment]
Superseded by #1164.

**Pull request closed without merging**